pkg/kp/pcstore/consul_store.go

```diff
@@ -193,6 +199,122 @@ func (s *consulStore) FindWhereLabeled(podID types.PodID,
 	return ret, nil
 }
 
+// Watch watches the entire podClusterTree for changes.
+// It will return a blocking channel on which the client can read
+// WatchedPodCluster objects. The goroutine maintaining the watch will block on
+// writing to this channel so it's up to the caller to read it with haste.
+func (s *consulStore) Watch(quit <-chan struct{}) <-chan WatchedPodCluster {
+	inCh := make(chan api.KVPairs)
+	outCh := make(chan WatchedPodCluster)
+	errChan := make(chan error, 1)
+
+	go consulutil.WatchPrefix(podClusterTree, s.kv, inCh, quit, errChan)
+
+	go func() {
+		var kvp api.KVPairs
+		for {
+			select {
+			case <-quit:
+				return
+			case err := <-errChan:
+				s.logger.WithError(err).Errorf("WatchPrefix returned error, recovered.")
+			case kvp = <-inCh:
+				if kvp == nil {
+					// nothing to do
+					continue
+				}
+			}
+
+			pcs, err := kvpsToPC(kvp)
+			if err != nil {
+				select {
+				case <-quit:
+					return
+				case outCh <- WatchedPodCluster{PodCluster: nil, Err: err}:
```

Comment:
might be nice to be explicit, but if `err != nil` then `pcs` is a 0 length slice so it's fine